### ISSUE TITLE
Tree Supports - Solid First Layer Joining All or Groups of Trees

### ISSUE THREAD

**Issue by neildmd** [Ultimaker/Cura#10340]
### Is your feature request related to a problem?

When using tree supports for large surface areas or models requiring many supports, Cura often generates many very small contact areas spread out across the silhouette of the surface to be supported for the first layer. This can lead to failed prints because the small tree features in contact with the bed do not adhere very well and are prone to getting knocked over or lifted up during the print.

### Describe the solution you'd like

It would be very useful if we had an option to print a solid first layer encompassing the silhouette of the area to be supported or perhaps encompassing the entire area where the tree supports meet the bed (having both options would be a bonus).

### Describe alternatives you've considered

I haven't considered any alternatives. Tree supports print faster and waste less filament than other support methods.

### Affected users and/or printers

All FDM 3D printer owners could benefit from this.

**Comment by fvrmr** [comment]
Could you share your project file (file - save project) with me? So I can take a look.

**Comment by neildmd** [comment]
Here you go fvrmr. [water dispenser v5.zip](https://github.com/Ultimaker/Cura/files/7102041/water.dispenser.v5.zip)

**Comment by Ghostkeeper** [comment]
It leaves these holes because there is a hard-coded minimum inner brim size. Without that limit, it would look like this:

![image](https://user-images.githubusercontent.com/2448634/132011650-0c490544-1323-4c85-a11f-bd4f14d40252.png)

With the limit:

![image](https://user-images.githubusercontent.com/2448634/132012028-f8aecff5-709b-419a-b6ac-01e7e295d1f7.png)

This limit is hard-coded to 100 square line widths. So the area of a square of 10x10 brim lines. This minimum is in place to prevent generating the centres of concentric brim lines which are places where excess material accumulates without anywhere to go. This accumulated plastic tends to pull the material off of the build plate, rather than adding to its adhesion.

The minimum area limit is only in place for holes. So the brim contours are only missing for parts between the supports, not for the supports themselves. Here is how it looks with 100% density to make the actual support areas visible:
![image](https://user-images.githubusercontent.com/2448634/132013392-0655f33c-c957-4653-85e1-f3815b9ee5e7.png)

I think we can all agree that if this were the shape of a model you don't want to have brim in those holes. You wouldn't want to get a tiny brim contour in the inside of every screw hole and such. They are detrimental to the adhesion.

The difference is that the support - in the case of tree support - gets printed with 0% density, so there is no adhesion coming from the inside of the shape. The filter of the limit applying only to holes kind of assumes that the contour is attached to the inside of the shape, which is a safe assumption if there is skin, but not if the infill density is 0% and there is no skin.

So you might consider this a setting issue. Not all settings work well with all models, after all. So the 0% density might have to be propped up in this case. The behaviour is expected in this case, but not very nice.

**Comment by neildmd** [comment]
@Ghostkeeper If you look closely at the 2nd image in my OP above you'll see that trees are being printed in mid air in some places @ layer 2 and even if they did print on layer 1 they would not be connected to the brim. I do understand your explanation and how this is a tricky situation but you are overlooking what seems to be a bug in this case. Here's a shot with 4 layers so you can see it easier:

![image](https://user-images.githubusercontent.com/6560233/133501932-cda1bb23-4af1-4e0a-9425-815b51137958.png)


Wouldn't it be rather easy to implement the feature I requested, where it just prints a solid layer in silhouette of the model? You could just adapt the raft setting to only print a single solid layer.

**Comment by neildmd** [comment]
Something else I just noticed, maybe I should make a separate bug report though. I can't open any 3mf files, standard or project type. It doesn't even show up in the list of supported extensions when attempting to open a file. I have 4.10.0 and 4.11.0 installed and neither will open any 3mf files.

**Comment by Ghostkeeper** [comment]
> If you look closely at the 2nd image in my post above you'll see that trees are being printed in mid air in some places @ layer 2 and even if they did print on layer 1 they would not be connected to the brim.

Yes, what I wanted to explain is that this happens because the shape that you see loose there is actually a hole in the support area. Since it removes small interior polygons of the brim (the minimum interior brim size mentioned before), it removes the brim around this wall, making it stand freely.

> Wouldn't it be rather easy to implement the feature I requested, where it just prints a solid layer in silhouette of the model?

It wouldn't solve the issue. Tree support expands to the side of the model, so the problem happens there too.

> I have 4.10.0 and 4.11.0 installed and neither will open any 3mf files.

Please report this separately. It could be that the 3MF reader plug-in has been disabled? A log file would be necessary.
